operators/freshdesk_to_s3_operator.py

```diff
@@ -0,0 +1,145 @@
+                                    credentials.
+    :type freshdesk_conn_id:        string
+    :param freshdesk_endpoint:      The endpoint to retrive data from.
+                                    Implemented for: 
+                                        - agents
+                                        - companies
+                                        - contacts
+                                        - conversations 
+                                        - groups
+                                        - roles
+                                        - satisfaction_ratings
+                                        - tickets
+                                        - time_entries
+    :type freshdesk_endpoint:       string
+    :param s3_conn_id:              The Airflow connection id used to store
+                                    the S3 credentials.
+    :type s3_conn_id:               string
+    :param s3_bucket:               The S3 bucket to be used to store
+                                    the Marketo data.
+    :type s3_bucket:                string
+    :param s3_key:                  The S3 key to be used to store
+                                    the Marketo data.
+    :type s3_bucket:                string
+    :param updated_at:              replication key
+    :type updated_at:               string
+    """
+
+    def __init__(self,
+                 freshdesk_conn_id,
+                 freshdesk_endpoint,
+                 s3_conn_id,
+                 s3_bucket,
+                 s3_key,
+                 updated_at=None,
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.freshdesk_conn_id = freshdesk_conn_id
+        self.freshdesk_endpoint = freshdesk_endpoint
+   
+        self.updated_at = parse(updated_at) if updated_at else None
+
+        self.s3_conn_id = s3_conn_id
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+
+        self.hook = None
+
+    def get_all(self, endpoint):
+        """
+        Get all pages.
+        """
+        response = self.hook.run(endpoint)
+        results = response.json()
+
+        while 'link' in response.headers:
+            response = self.hook.run(response.headers['link'])
+            results.extend(response.json())
+        # filter the results 
+        if self.updated_at:
+          results = [x for x in results
+                     if (not 'updated_at' in x) or
+                     (parse(x['updated_at']).replace(tzinfo=None) > self.updated_at)]
+
+
+        return results
+
+    def execute(self, context):
+        """
+        Execute the operator.
+        This will get all the data for a particular Freshdesk model
+        and write it to a file.
+        """
+        logging.info("Prepping to gather data from Freshdesk")
+        self.hook = FreshdeskHook(
+            http_conn_id=self.freshdesk_conn_id
```

Comment:
I think this will look better if it will go on a single line.

Reply:
Now I have 2 args here, but good to know^^